qupath-core/src/main/java/qupath/lib/common/GeneralTools.java

```diff
@@ -757,14 +758,20 @@ public static boolean isWindows() {
 
 
 	/**
-	 * Delete a file, optionally requesting that it be moved to the trash rather than permanently deleted.
+	 * Delete a file or an empty directory, optionally requesting that it be moved to the trash rather than permanently deleted.
 	 * <p>
-	 * Note that the behavior of this method is system-dependent, and there is no guarantee the file will 
+	 * Note that the behavior of this method is system-dependent, and there is no guarantee the file/empty directory will
 	 * indeed be moved to the trash.
 	 * 
-	 * @param fileToDelete
-	 * @param preferTrash
-	 * @return true if the file is successfully deleted, false otherwise
+	 * @param fileToDelete the file or empty directory to delete
+	 * @param preferTrash whether an attempt should be made to move the provided file/empty directory to trash before deleting it
+	 * @return true if the provided file/empty directory is successfully deleted, false otherwise
+	 * @throws java.awt.HeadlessException if the preferTrash parameter and {@link GraphicsEnvironment#isHeadless()} are true
```

Comment:
I don't think the fact we are using `Desktop` should be exposed to the caller. I would tend towards throwing fewer exceptions and using the return value to handle failures - logging the reason (although I think NPE and `IllegalArgumentException` make sense to keep).